src/javascript/app_2/Stores/Modules/Trading/Helpers/proposal.js

```diff
@@ -38,12 +38,19 @@ export const createProposalRequests = (store) => {
 
 const createProposalRequestForContract = (store, type_of_contract) => {
     const obj_expiry = {};
+
     if (store.expiry_type === 'endtime') {
         const expiry_date = moment.utc(store.expiry_date);
         const start_date  = moment.unix(store.start_date || (store.root_store.common.server_time / 1000)).utc();
         const is_same_day = expiry_date.isSame(start_date, 'day');
-        const expiry_time = is_same_day ? store.expiry_time : '23:59:59';
-        obj_expiry.date_expiry = convertToUnix(expiry_date.unix(), expiry_time);
+
+        if (store.contract_start_type === 'spot') {
+            const expiry_time = store.expiry_time;
+            obj_expiry.date_expiry = convertToUnix(expiry_date.unix(), expiry_time);
+        } else {
+            const expiry_time = is_same_day ? store.expiry_time : moment().utc().format('HH:mm');
+            obj_expiry.date_expiry = convertToUnix(expiry_date.unix(), expiry_time);
```

Comment:
seems this line is just repeated in both if and else? if so it can be moved outside and set only once